spec/views/availability_zone/show.html.haml_spec.rb

```diff
@@ -1,15 +1,16 @@
 describe "availability_zone/show.html.haml" do
   shared_examples_for "miq_before_onload JS is needed" do
     it "renders proper JS" do
-      js_string = "ManageIQ.afterOnload = {action: () => miqAsyncAjax(\"/availability_zone/#{action}/#{availability_zone.id}\")}"
+      template_path = "availability_zone/#{action}"
+      js_string = "ManageIQ.afterOnload = {action: () => miqAsyncAjax(\"/#{template_path}/#{availability_zone.id}\")}"
       
-      render
+      render :template => template_path
```

Comment:
renders without a template is raising the missing template error and the bulk of the changes in this PR.